### PR TITLE
Bug 1986437: Bump github.com/openshift/api

### PR DESCRIPTION
**Pull request by Fedosin**
This change includes an important update for ExternalCloudProvider feature gate: https://github.com/openshift/api/pull/961

Generated by `go get -u github.com/openshift/api` + `go mod vendor`

### CODE REVIEW

**Comment by elmiko** [comment]
+1, this is needed to help the external cloud controller manager effort

**Comment by sinnykumari** [comment]
You need to update openshift/api version at https://github.com/openshift/machine-config-operator/blob/master/go.mod#L70

Q. There seems to be lot of deps update, are they all result of just updating openshift/api to 20210720160326-96bb0f993a66 ?

**Comment by Fedosin** [comment]
@sinnykumari unfortunately now one of our dependencies is broken (klog) For more information: https://github.com/kubernetes/klog/issues/253 
So, this PR does two things: first it fixes the issue with klog, and then bumps openshift/api. For clarity I can split this PR into two commits.

**Comment by Fedosin** [comment]
Without additional changes it looks like this:

```sh
❯ go get -u github.com/openshift/api
# k8s.io/klog/v2
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:705:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:724:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:742:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:763:10: invalid operation: logr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:782:11: invalid operation: loggr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:783:3: undefined: logr.WithCallDepth
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:794:11: invalid operation: loggr != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:795:3: undefined: logr.WithCallDepth
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:915:9: invalid operation: log != nil (mismatched types logr.Logger and nil)
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:919:4: undefined: logr.WithCallDepth
../../go/pkg/mod/k8s.io/klog/v2@v2.10.0/klog.go:919:4: too many errors
```

**Comment by sinnykumari** [comment]
> @sinnykumari unfortunately now one of our dependencies is broken (klog) For more information: [kubernetes/klog#253](https://github.com/kubernetes/klog/issues/253)
> So, this PR does two things: first it fixes the issue with klog, and then bumps openshift/api. For clarity I can split this PR into two commits.

yes, splitting the commit makes more sense here.

**Comment by kikisdeliveryservice** [comment]
Q: do we have anything tracking: https://github.com/kubernetes/klog/pull/242 to remove that commit

**Comment by sinnykumari** [comment]
In go.mod, you haven't updated `replace` section with updated versions of deps and as a result vendor file for deps like [openshift/api](https://github.com/openshift/machine-config-operator/pull/2685/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L70), k8s.io/api, etc has not been pulled in. You would want to update deps mentioned in replace section so that it doesn't replace back to old one.

**Comment by sinnykumari** [comment]
I think you are getting confused when updating replace section. You should only update deps version in `replace` section of go.mod that actually has corresponding changes in this PR in require section of go.mod. See https://golang.org/ref/mod#go-mod-file-replace

File: go.mod [hunk 1]
```diff
@@ -69,29 +71,33 @@ replace (
...
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.21.3
+	k8s.io/metrics => k8s.io/metrics v0.21.3
 	k8s.io/mount-utils => k8s.io/mount-utils v0.23.0-alpha.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.0-rc.0
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.3
 )
+
+// Workaround to deal with https://github.com/kubernetes/klog/issues/253
+// Should be deleted when https://github.com/kubernetes/klog/pull/242 is merged
+exclude github.com/go-logr/logr v1.0.0
```
Comment (aojea) [inline go.mod:103]: it worked for me adding it as a `replace`

```
github.com/go-logr/logr =>  github.com/go-logr/logr v0.4.0
```

Comment (Fedosin) [reply go.mod:104]: The "official" recommendation is to use `exclude` https://github.com/kubernetes/klog/issues/253#issuecomment-883465311 but I can try with replace.

Comment (Fedosin) [reply go.mod]: done

Comment (aojea) [reply go.mod]: I trust dims more than me, just use the one you want meanwhile unblocks you :smile:

**Review by sinnykumari — APPROVED** [review]
LGTM
/retest

**Pull request merged**